dojo/finding_group/redis.py

```diff
@@ -0,0 +1,248 @@
+import base64
+import json
+import logging
+from dataclasses import asdict, dataclass, field
+from datetime import datetime
+from enum import StrEnum
+from functools import lru_cache
+from typing import Optional
+
+import redis
+from django.utils.safestring import mark_safe
+
+from dojo.models import Finding
+
+logger = logging.getLogger(__name__)
+
+SEVERITY_ORDER = {
+    "Critical": 5,
+    "High": 4,
+    "Medium": 3,
+    "Low": 2,
+    "Info": 1,
+}
+USER_MODES_KEY = "finding_groups_user_modes"
+SYSTEM_CHANGE = "finding_groups_last_finding_change"
+LAST_UPDATE = "finding_groups_last_update"
+
+
+class GroupMode(StrEnum):
+    VULN_ID_FROM_TOOL = "vuln_id_from_tool"
+    TITLE = "title"
+    CVE = "cve"
```

Comment:
We should write some documentation explaining how other `GroupMode`s can be added.